src/Validation/RequestValidator.php

```diff
@@ -122,7 +122,22 @@ protected function validateParameters()
 
                 if ($actual_parameter) {
                     if ($expected_parameter_schema->type && gettype($actual_parameter) !== $expected_parameter_schema->type) {
-                        settype($actual_parameter, $expected_parameter_schema->type);
+                        $typeIsCorrect = true;
+                        switch ($expected_parameter_schema->type){
+                            case 'integer':
+                            case 'number':
+                                $typeIsCorrect = is_numeric($actual_parameter);
+                                break;
+                            case 'string':
+                                $typeIsCorrect = is_string($actual_parameter);
+                                break;
+                            case 'boolean':
+                                $typeIsCorrect = filter_var($actual_parameter, FILTER_VALIDATE_BOOLEAN) !== false;
```

Comment:
in openapi, booleans are `'true'` and `'false'` only
(https://swagger.io/docs/specification/data-models/data-types/#boolean, https://spec.openapis.org/oas/v3.1.0#data-types)
Using `filter_var` here can lead to false positive as `'on'` will be considered as valid boolean (which is not)